sqlglot/dialects/tsql.py

```diff
@@ -719,9 +719,19 @@ def _parse_user_defined_function(
             ):
                 return this
 
-            expressions = self._parse_csv(self._parse_function_parameter)
+            expressions = None
+            if self._curr.token_type != TokenType.WITH:
+                expressions = self._parse_csv(self._parse_function_parameter)
```

Comment:
```suggestion
            if self._match(TokenType.WITH, advance=False):
                expressions = self._parse_csv(self._parse_function_parameter)
            else:
                expressions = None
```

Reply:
I'll make this change

Reply:
The line `if self._match(TokenType.WITH, advance=False):` is missing a `not` since I want to check that the current token is not WITH. I will add that on my next push.